lib/metasploit/framework/compiler/mingw.rb

```diff
@@ -43,90 +47,159 @@ def build_cmd(src)
           cmd << '-fno-ident '
           cmd << opt_level
 
-          if self.compile_options
-            cmd << self.compile_options
+          if compile_options
+            cmd << compile_options
           else
             link_options << '--image-base=0x0,'
             cmd << '-nostdlib '
           end
 
           link_options << '--no-seh'
-          link_options << ',-s' if self.strip_syms
-          link_options << ",-T#{self.link_script}" if self.link_script
+          link_options << ',-s' if strip_syms
+          link_options << ",-T #{link_script}" if link_script
 
           cmd << link_options
-
           cmd
         end
 
-        def cleanup_files
-          src_file = "#{self.file_name}.c"
-          exe_file = "#{self.file_name}.exe"
+        def compile_cpp(src, *additional_files)
+          within_temp(additional_files) do |dir, fnames|
+            file = Tempfile.create(['', '.cpp'], dir)
+            File.write(file, src)
+            if keep_src
+              dest = File.join(File.dirname(outfile), File.basename(file.path))
+              FileUtils.cp(file, dest)
+            end
+            fnames << file.path
+            cmd = build_cpp_files_cmd(fnames)
+            return exec(cmd), dest
+          end
+        end
+
+        def compile_cpp_files(files)
+          within_temp(files) do |_dir, fnames|
+            cmd = build_cpp_files_cmd(fnames)
+            exec(cmd)
+          end
+        end
+
+        def compile_cpp_file(file)
+          compile_cpp_files([file])
+        end
```

Comment:
Not going to lie, this seems a little pointless when the user can just pass a single entry array themselves. I get trying to create a helper function but this doesn't really seem to do much that the user can't just do on their own and only serves to add another function to the function call chain, meaning your just adding more overhead to the program with minimal benefits.

Reply:
it's pretty conventional, even within Rails' source code
the overhead of a single call stack frame is infinitesimally small

Reply:
pointless method go boom